src/data-table/measure-column-widths.js

```diff
@@ -12,46 +12,60 @@ import {useStyletron} from '../styles/index.js';
 
 import HeaderCell from './header-cell.js';
 import type {ColumnT, RowT} from './types.js';
+import {useRef} from 'react';
 
-// https://github.com/Swizec/useDimensions
-function useDimensions() {
-  const [dimensions, setDimensions] = React.useState({});
-  const [node, setNode] = React.useState(null);
+// Measures the column header + sampled data
+function MeasureColumn({sampleIndexes, column, columnIndex, rows, onLayout}) {
+  const [css] = useStyletron();
 
-  const ref = React.useCallback(node => {
-    setNode(node);
-  }, []);
+  const ref = useRef();
 
   React.useEffect(() => {
-    if (__BROWSER__) {
-      if (node) {
-        window.requestAnimationFrame(() => {
-          setDimensions(node.getBoundingClientRect());
-        });
-      }
+    if (ref.current) {
+      onLayout(columnIndex, ref.current.getBoundingClientRect());
     }
-  }, [node]);
-
-  return [ref, dimensions];
-}
-
-type ElementMeasurerPropsT = {
-  onDimensionsChange: (dimensions: {width: number}) => void,
-  // eslint-disable-next-line flowtype/no-weak-types
-  item: React.Element<any>,
-};
-
-function ElementMeasurer(props: ElementMeasurerPropsT) {
-  const {onDimensionsChange} = props;
-  const [ref, dimensions] = useDimensions();
-
-  React.useEffect(() => {
-    onDimensionsChange(dimensions);
-  }, [dimensions, onDimensionsChange]);
+  }, []);
 
-  return React.cloneElement(props.item, {ref});
+  return (
+    <div
```

Comment:
Awesome! just measuring the parent is such a painfully obvious improvement. Thank you